### PR TITLE
mergify: add config from flux-core

### LINKED ISSUE #559: Add mergify bot to flux-sched
Any opposition to adding the mergify bot to this repo?  It is the same one that is used in flux-core to help with PR merging.  Once you set the 'merge-when-passing' label on a PR, mergify automatically attempts to rebase PRs onto master (assuming the PR branch is out-of-date), and if the rebase is conflict-free and passes Travis, mergify updates the PR and merges it.

### PR DESCRIPTION
**Pull request by SteVwonder**
Closes #559

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+pull_request_rules:
+  - name: rebase and merge when passing all checks
+    conditions:
+      - base=master
+      - status-success=continuous-integration/travis-ci/pr
+      - status-success="validate commits"
+      - label="merge-when-passing"
+      - label!="work-in-progress"
+      - "approved-reviews-by=@flux-framework/core"
```
Comment (dongahn) [inline .mergify.yml]: Assume @flux-framework/core covers all of the core LLNL developers?

**Comment by dongahn** [comment]
LGTM. I just have one question for my own edification.

File: .mergify.yml (hunk 1 above)
Comment (SteVwonder) [reply .mergify.yml]: Yep!

**Comment by dongahn** [comment]
@SteVwonder: Could you rebase? Then this will go in next.

**Comment by SteVwonder** [comment]
> @SteVwonder: Could you rebase? Then this will go in next.

Great! Rebased onto master.  Once it is merged, I (or someone else) will need to enable mergify on the repo (just not before the config is merged, I almost learned the hard way on that one).

**Comment by dongahn** [comment]
Got it. Thanks!

**Comment by grondo** [comment]
I added mergify.io to flux-sched just now. Thanks!

**Pull request merged**